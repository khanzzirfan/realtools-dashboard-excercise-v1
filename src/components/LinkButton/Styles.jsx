import { Link } from 'react-router-dom';
import styled from 'styled-components';

export const StyledLink = styled(Link)`
  text-decoration: none;
  color: var(--colors-irisBlue);
  font-weight: 500;
  font-size: 1rem;
  line-height: 19px;
`;
