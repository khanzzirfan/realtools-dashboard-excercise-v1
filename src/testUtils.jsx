import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { QueryClient, QueryClientProvider } from 'react-query';
import { MemoryRouter } from 'react-router-dom';
import { AppContextProvider } from './hooks/useAppContext';
import { theme } from './theme';

const AllTheProviders = ({ children }) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        // âœ… turns retries off
        retry: 2,
      },
    },
  });
  return (
    <QueryClientProvider client={queryClient}>
      <AppContextProvider>
        <ThemeProvider theme={theme}>
          <MemoryRouter initialEntries={['/people', '/people/new', '/people/edit/:id']}>
            {children}
          </MemoryRouter>
        </ThemeProvider>
      </AppContextProvider>
    </QueryClientProvider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
