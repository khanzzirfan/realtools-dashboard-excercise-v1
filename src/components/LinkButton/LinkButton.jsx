import React from 'react';
import PropTypes from 'prop-types';
import { StyledLink } from './Styles';

const LinkButton = ({ children, to, ...props }) => {
  return (
    <StyledLink to={to} {...props}>
      {children}
    </StyledLink>
  );
};

LinkButton.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.node]).isRequired,
};

export default LinkButton;
