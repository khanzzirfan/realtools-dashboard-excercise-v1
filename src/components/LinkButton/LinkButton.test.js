import { axe } from 'jest-axe';
import { render } from '../../testUtils';
import LinkButton from './LinkButton';

describe('LinkButton', () => {
  it('renders correctly with default styles', () => {
    const { asFragment } = render(<LinkButton to="/people/edit/1">Edit</LinkButton>);
    expect(asFragment()).toMatchSnapshot();
  });

  /** axe voilations check */
  it('should not have axe voilations', async () => {
    const { container } = render(<LinkButton to="/people/edit/1">Edit</LinkButton>);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
