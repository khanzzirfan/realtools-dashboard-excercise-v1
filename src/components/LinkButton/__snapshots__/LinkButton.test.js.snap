// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton renders correctly with default styles 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: var(--colors-irisBlue);
  font-weight: 500;
  font-size: 1rem;
  line-height: 19px;
}

<a
    class="c0"
    href="/people/edit/1"
  >
    Edit
  </a>
</DocumentFragment>
`;
